pkg/gce-pd-csi-driver/device_error_map.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gceGCEDriver
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// deviceErrMap is an atomic data datastructure for recording deviceInUseError times
+// for specified devices
+type deviceErrMap struct {
+	mux               sync.Mutex
+	deviceInUseErrors map[string]time.Time
+}
+
+var DeviceInUseTimeout = time.Second * 30
```

Comment:
Can you set this as a field of `deviceErrMap` and plumb it through constructor instantiation? I'd rather have this set as a field on the Node server, rather than initialized as a global variable. Global variables limit the ability of this logic to be unit tested, and limit concurrency patterns if the driver is refactored.

Reply:
Yea good suggestion. I ended up plumping it through to the `deviceInUseErrors` struct, as the logic in NodeServer itself doesn't really care about it.